### PR TITLE
RSDK-1851 Backend implementation for removing Interaction Spaces from WorldState

### PR DESCRIPTION
**Pull request by nfranczak**

### CODE REVIEW

**Review by raybjork — APPROVED** [review]

**Review by raybjork — CHANGES_REQUESTED** [review]
Just realized that this PR can't be merged until we scrub motion-testing of interaction spaces.  Otherwise motion CI tests will continue to fail

**Comment by nfranczak** [comment]
> Just realized that this PR can't be merged until we scrub motion-testing of interaction spaces. Otherwise motion CI tests will continue to fail

adding jira ticket now. Will attach link shortly

File: motionplan/constraint.go [hunk 1]
```diff
@@ -146,11 +146,7 @@ func NewCollisionConstraint(
 	if err != nil {
 		return nil
 	}
-	spaceEntities, err := NewSpaceCollisionEntities(interactionSpaces)
```
Comment (biotinker) [inline motionplan/constraint.go:149]: This is an opportunity to make our collision system massively more readable.

We can take now as an opportunity to rip out `SpaceCollisionEntities` entirely.

Comment (spieswl) [reply motionplan/constraint.go:149]: Completely agreed. That tripped me up a bunch when looking through this initially. Should just be `CollisionEntities` for all incarnations of the concept.

**Review by spieswl — CHANGES_REQUESTED** [review]
Note Peter's suggestion on tidying up some content in `collision.go` is a good one and we should follow up on that

**Review by spieswl — APPROVED** [review]
Changing my mind on the status. @biotinker and I talked a little bit and we agreed that changes to `collision.go` are probably not a good use of this PR. So changing my approval status to reflect the new resolution.

**Review by biotinker — APPROVED** [review]
This LGTM

**Comment by spieswl** [comment]
Also note that I re-ran the Motion Benchmark step, which completed successfully.

**Pull request merged**